cli/internal/cloudcmd/create.go

```diff
@@ -25,28 +23,21 @@ import (
 
 // Creator creates cloud resources.
 type Creator struct {
-	out            io.Writer
-	newGCPClient   func(ctx context.Context, project, zone, region, name string) (gcpclient, error)
-	newAzureClient func(subscriptionID, tenantID, name, location, resourceGroup string) (azureclient, error)
-	newQEMUClient  func(ctx context.Context) (qemuclient, error)
+	out                io.Writer
+	newTerraformClient func(ctx context.Context) (terraformClient, error)
+	newAzureClient     func(subscriptionID, tenantID, name, location, resourceGroup string) (azureclient, error)
 }
 
 // NewCreator creates a new creator.
 func NewCreator(out io.Writer) *Creator {
 	return &Creator{
 		out: out,
-		newGCPClient: func(ctx context.Context, project, zone, region, name string) (gcpclient, error) {
-			return gcpcl.NewInitialized(ctx, project, zone, region, name)
+		newTerraformClient: func(ctx context.Context) (terraformClient, error) {
+			return terraform.New(ctx, cloudprovider.GCP)
```

Comment:
This breaks QEMU and other CSPs. The cloud provider should be passed as a variable to the function.